### PR TITLE
Add additional data cleaning based on checkpoint 1 feedback

### PR DESCRIPTION
**Pull request by chanteriam**
## Describe your changes
I cleaned and filtered the corpus data based on the following criteria (in datacleaner.py):
- only included cases from 2014-2019 (the latest year of the dataset)
- only included cases where the winning side was either 0 or 1 (no missing, etc.)
- filtered the speakers, conversations, utterances, voters, and advocates dataframes based on the case_ids returned from the above cleaning and filtering

In descriptives.py, I edited the file based on the new dataframes returned above and also included more descriptive statistics based on the voting habits of particular justices.

Finally, I added a make command for tokenization.

## Checklist before requesting a review

- [x] The code runs successfully.
- [x] I ran `make lint` and have made the bulk of changes that it has requested.

### CODE REVIEW

File: supreme_court_predictions/statistics/descriptives.py [hunk 1]
```diff
@@ -239,15 +232,42 @@ def get_voters_desc(self):
...
+        for justice in justices:
+            # get counts
+            voters_df.loc[("justice", justice), "counts"] = len(
+                voters.loc[voters.loc[:, "voter"] == justice, :]
+            )
+
+            # get percent for petitioner
+            voters_df.loc[("justice", justice), "percentages"] = voters.loc[
+                voters.loc[:, "voter"] == justice, "vote"
+            ].sum() / len(voters.loc[voters.loc[:, "voter"] == justice, :])
```
Comment (JJModern) [inline supreme_court_predictions/statistics/descriptives.py:267]: great to see

File: supreme_court_predictions/statistics/datacleaner.py [hunk 2]
```diff
@@ -126,59 +204,76 @@ def get_conversation_dfs(conversations_dict):
...
+                if voters:
+                    for voter, vote in voters.items():
+                        if vote in [0, 1]:
+                            vote_dict = {
+                                "id": conversation_id,
+                                "case_id": conversation_data["case_id"],
+                                "voter": voter,
+                                "vote": vote,
+                            }
+                            voters_list.append(vote_dict)
```
Comment (JJModern) [inline supreme_court_predictions/statistics/datacleaner.py:239]: big brain

**Review by JJModern — APPROVED** [review]

File: supreme_court_predictions/__main__.py [hunk 3]
```diff
@@ -39,6 +40,14 @@
         action=argparse.BooleanOptionalAction,
     )
 
+    parser.add_argument(
+        "--tokenize-data",
+        help="Generate tokenizations for Convokit",
+        type=bool,
+        default=False,
+        action=argparse.BooleanOptionalAction,
+    )
```
Comment (michplunkett) [inline supreme_court_predictions/__main__.py:49]: So on it.

File: supreme_court_predictions/statistics/datacleaner.py [hunk 4]
```diff
@@ -12,6 +12,7 @@
 from supreme_court_predictions.util.contants import (
     ENCODING_UTF_8,
     FILE_MODE_READ,
+    LATEST_YEAR,
```
Comment (michplunkett) [inline supreme_court_predictions/statistics/datacleaner.py:15]: Noice.

File: Makefile [hunk 5]
```diff
@@ -35,3 +35,7 @@ clean-data:
 .PHONY: describe-data
 describe-data:
 	python -m ${BASEDIR} --describe-data
+
+.PHONY: tokenize-data
+tokenize-data:
+	python -m ${BASEDIR} --tokenize-data
```
Comment (michplunkett) [inline Makefile:41]: Add this to the `run` command above, plz.

File: supreme_court_predictions/__main__.py [hunk 6]
```diff
@@ -49,3 +58,6 @@
 
     if args.describe_data:
         describe_data()
+
+    if args.tokenize_data:
+        tokenize_data()
```
Comment (michplunkett) [inline supreme_court_predictions/__main__.py:63]: I think you'll need to move this above so the token data is there, should it be used in the describe function.

File: supreme_court_predictions/statistics/descriptives.py [hunk 7]
```diff
@@ -239,15 +232,42 @@ def get_voters_desc(self):
...
 
         voter_stats.loc[("justices", ""), "counts"] = len(
             voters.loc[:, "voter"].unique()
         )
 
+        voter_stats = pd.concat([voter_stats, self.votes_by_justice(voters)])
+
         return voter_stats
 
+    def votes_by_justice(self, voters):
```
Comment (michplunkett) [inline supreme_court_predictions/statistics/descriptives.py:246]: Needs a `:param:` thang.

**Review by michplunkett — CHANGES_REQUESTED** [review]
Some smöl changes.

File: Makefile (hunk 5 above)
Comment (chanteriam) [reply Makefile:41]: u right

File: supreme_court_predictions/__main__.py (hunk 6 above)
Comment (chanteriam) [reply supreme_court_predictions/__main__.py:63]: gotcha

File: supreme_court_predictions/__main__.py [hunk 8]
```diff
@@ -31,6 +32,14 @@
         action=argparse.BooleanOptionalAction,
     )
 
+    parser.add_argument(
+        "--tokenize-data",
+        help="Generate tokenizations for Convokit",
+        type=bool,
+        default=False,
+        action=argparse.BooleanOptionalAction,
+    )
```
Comment (michplunkett) [inline supreme_court_predictions/__main__.py:41]: The order of these don't matter, but it makes sense to me.

**Review by michplunkett — APPROVED** [review]
Looks great! Well done, @chanteriam! 🧠 🧠 🧠 🧠 🧠

**Pull request merged**